api/capi/src/nntrainer.cpp

```diff
@@ -968,6 +968,84 @@ int ml_train_dataset_destroy(ml_train_dataset_h dataset) {
   return status;
 }
 
+int ml_train_model_get_input_dimensions(ml_train_model_h model,
+                                        ml_tensors_info_h *info) {
+  int status = ML_ERROR_NONE;
+  ml_train_model *nnmodel;
+  std::shared_ptr<ml::train::Model> m;
+  returnable f;
+
+  check_feature_state();
+
+  ML_TRAIN_GET_VALID_MODEL_LOCKED(nnmodel, model);
+  ML_TRAIN_ADOPT_LOCK(nnmodel, model_lock);
+  m = nnmodel->model;
+
+  std::vector<nntrainer::TensorDim> dims = m->getInputDimension();
+
+  f = [&]() { return ml_tensors_info_set_count(*info, dims.size()); };
```

Comment:
where do you allocate `info`? Also please clarify in the header documentation if the info must be allocated by the caller or the callee (based on the function signature, I am assuming its callee).

Reply:
you don't need to wrap `ml_tensors_info_*` functions in `nntrainer_exception_boundary`? They are c-API functions and they will not throw, just return error code.

Reply:
Done